lib/tree.js

```diff
@@ -7,18 +7,36 @@ var metadata = require('./helpers/metadata');
 function tree(opts) {
   opts = _.defaults(opts || {}, {
     deep: false,
+    depth: null,
   });
 
   var tasks = this._registry.tasks();
-  var nodes = _.map(tasks, function(task) {
-    var meta = metadata.get(task);
 
-    if (opts.deep) {
-      return meta.tree;
-    }
-
-    return meta.tree.label;
-  });
+  var nodes;
+  if (!opts.deep) {
```

Comment:
Is it ok that `opts.deep` should be removed and show all depth when `opts.depth == null || opts.depth == Infinity` ?